game/gameboard.py

```diff
@@ -242,9 +242,8 @@ def _validate_matrix(self, matrix_to_test):
         if matrix_to_test.shape != (6,7):
             return False
         for row in matrix_to_test:
-            for entry in row:
-                if not self._is_valid_cell_value(entry):
-                    return False
+            if not all([self._is_valid_cell_value(entry) for entry in row]):
```

Comment:
No need for list comprehension :) http://stackoverflow.com/questions/405516/if-all-in-list-something